sdk/client-core/core/src/main/java/io/clientcore/core/http/models/RequestOptions.java

```diff
@@ -375,6 +376,30 @@ public RequestOptions setLogger(ClientLogger logger) {
         return this;
     }
 
+
+    /**
+     * Get the specified event stream {@link ServerSentEventListener listener} for this request.
+     *
+     * @return The {@link ServerSentEventListener listener} for this request.
+     */
+    public ServerSentEventListener getServerSentEventListener() {
+        return serverSentEventListener;
+    }
+
+    /**
+     * Set an event stream {@link ServerSentEventListener listener} for this request.
+     *
+     * @param serverSentEventListener The {@link ServerSentEventListener listener} to set for this request.
+     *
+     * @return The updated {@link RequestOptions}.
+     */
+    public RequestOptions setServerSentEventListener(ServerSentEventListener serverSentEventListener) {
+        this.serverSentEventListener = serverSentEventListener;
+
+        return this;
+    }
```

Comment:
Just so I understand - if a user does not set this in a SSE method, I presume they will not get any value from calling that method? It helps me understand if we should have this be so hidden in the `RequestOptions` type, or if we either need to elevate it out of it, or make a subtype of it.

Reply:
> if a user does not set this in a SSE method, I presume they will not get any value from calling that method

They will get a Response with deserialized data values but no call back functionality, yes.

Reply:
I wonder if we should elevate this out of RequestOptions then and have it be a required parameter on client methods that are SSE?

Reply:
Yeah, we should just make it required param for SSE data operations. It wouldn't be much of use for the `text\event-stream` type to not have registered callbacks.

Reply:
Put it back on the `HttpRequest` to have it plumb it through RestProxy.